### PR TITLE
[DPR2-1514] Fix AWS library mismatch issue and use STS with DDB

### PR DESCRIPTION
**Pull request by stalwartstu**
- Promoted STS config from Main API to Library (to be used optionally).
- Added use of STS to DDB.
- Switched from Kotlin to Java version of DDB AWS SDK.
- Removed now-unnecessary suspend qualifiers on functions.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportinglib/config/AthenaConfig.kt [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import software.amazon.awssdk.services.athena.AthenaClient
+import software.amazon.awssdk.services.sts.auth.StsAssumeRoleCredentialsProvider
+
+@Configuration
+class AthenaConfig {
+
+  @Bean
```
Comment (gavriil-g-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportinglib/config/AthenaConfig.kt:13]: Could we make these conditional on missingBean?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportinglib/config/AthenaConfig.kt [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  @ConditionalOnBean(StsAssumeRoleCredentialsProvider::class)
+  fun athenaClientSts(
+    stsAssumeRoleCredentialsProvider: StsAssumeRoleCredentialsProvider,
+    properties: AwsProperties,
+  ): AthenaClient = AthenaClient.builder()
+    .region(properties.typedRegion)
+    .credentialsProvider(stsAssumeRoleCredentialsProvider)
+    .build()
+
+  @Bean
```
Comment (gavriil-g-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportinglib/config/AthenaConfig.kt:24]: (https://github.com/ministryofjustice/hmpps-digital-prison-reporting-lib/pull/499/files#r1880078969)

**Review by gavriil-g-moj — DISMISSED** [review]
We agreed to make the async related Beans conditional. 
It can be either as part of this PR or as a next change.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportinglib/data/DynamoDbProductDefinitionRepository.kt [hunk 3]
```diff
@@ -1,36 +1,37 @@
...
+        .indexName(properties.dynamoDb.categoryIndexName)
+        .keyConditionExpression("${properties.dynamoDb.categoryFieldName} = :${properties.dynamoDb.categoryFieldName}")
+        .expressionAttributeValues(attrValues)
+        .build()
     }
   }
 
-  override suspend fun getProductDefinitions(path: String?): List<ProductDefinition> {
-    return dynamoDbClient.query(getQueryRequest(properties, path ?: defaultPath)).items
-      ?.filter { it[properties.definitionFieldName] != null }
```
[the remark this answers is not in the document]
Comment (stalwartstu) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportinglib/data/DynamoDbProductDefinitionRepository.kt:32]: Oops, that wasn't on purpose. Good catch!

**Review by gavriil-g-moj — APPROVED** [review]

**Pull request merged**